quinn-proto/src/connection.rs

```diff
@@ -1413,6 +1428,15 @@ impl Connection {
         Ok(())
     }
 
+    fn send_ready_events(&mut self) {
+        if self.side.is_server() {
+            self.endpoint_events.push_back(EndpointEvent::HandshakeEnded);
+        }
+        if self.config.local_cid_len != 0 && !self.state.is_closed() {
+            self.endpoint_events.push_back(EndpointEvent::IdentifiersNeeded);
```

Comment:
Instead of having a separate case for this, we could rely on `issue_cid` being a harmless no-op on a closed connection and key off `HandshakeEnded` alone, but I'm okay with this approach if you prefer it.